io/disk/rawread.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2019 IBM.
+# Author: Naresh Bannoth <nbannoth@in.ibm.com>
+
+"""
+Rawread test
+"""
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import archive
+from avocado.utils import build
+from avocado.utils import process, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Rawread(Test):
+
+    """
+    Rawread is a benchmark suite that is aimed at performing a number
+    of simple tests of hard drive like write and read
+    """
+
+    def setUp(self):
+        """
+        checking install of required packages and extract and
+        compile of rawread suit.
+        """
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'libaio-devel']
+        if distro.detect().name == 'Ubuntu':
+            deps.extend(['g++'])
+        else:
+            deps.extend(['gcc-c++'])
+
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel("Fail to install Package: %s" % package)
+
+        tarball_path = os.path.join(os.path.abspath(''),
+                                    "rawread.py.data/rawread.tar")
+        tarball = tarball_path.split("/")[-1]
+        archive.extract(tarball_path, self.teststmpdir)
+        self.source = os.path.join(self.teststmpdir,
+                                   os.path.basename(
+                                       tarball.split('.tar')[0]))
+        os.chdir(self.source)
+        build.make(self.source)
+
+        self.disk = self.params.get('disk', default=None)
+
+    def test(self):
+        """
+        Run 'rawread' with its arguments
+        """
+        for val in range(23):
+            cmd = "./rawread -t %s %s " % (val, self.disk)
+            if process.system(cmd, shell=True, ignore_status=True):
+                self.fail("test failed")
```

Comment:
Will a fail in val=5 means it will fail for val=6 and higher values also ?

Reply:
if the command fail, then it will fail there itself before going to next value.

Reply:
@Naresh-ibm I understand that from the code. I was asking the behaviour of the test outside of automation

Reply:
NO, Command runs individually on given option at that instance. it does not depend on previous run result.

Reply:
@Naresh-ibm Let me ask my question in a better way.
Is it useful to do something like below, which is `fail last` ?
```
for
    if process.system(cmd, shell=True, ignore_status=True):
        err.append(val)
if err:
    self.fail(val)
```

Reply:
can be doable. but we have many more option which we have to explore and add them.
So we can change that time. if you want I can add it right away too.